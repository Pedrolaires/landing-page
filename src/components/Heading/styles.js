import styled, { css } from "styled-components";

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  large: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (upperCase) => css`
  text-transform: ${upperCase ? "uppercase" : "none"};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, upperCase }) => css`
    color: ${colorDark
      ? theme.colors.lightGrayColor
      : theme.colors.primaryColor};

    ${titleSize[size](theme)};
    ${titleCase(upperCase)};
  `}
`;
