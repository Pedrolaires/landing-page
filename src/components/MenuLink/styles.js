import styled, { css } from "styled-components";
import theme from "../../styles/theme";

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;
    background: blue;

    &::after {
      content: "";
      position: absolute;
      bottom: 0;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
