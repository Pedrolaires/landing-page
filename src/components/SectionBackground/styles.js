import styled, { css } from "styled-components";

const containerBackgroundActivate = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.whiteColor};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.whiteColor};
    color: ${theme.colors.primaryColor};
    ${background && containerBackgroundActivate(theme)};

    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
