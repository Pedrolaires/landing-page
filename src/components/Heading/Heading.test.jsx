import { screen } from "@testing-library/react";
import { Heading } from ".";
import { renderTheme } from "../../styles/render-theme";
import { theme } from "../../styles/theme";

describe("<Heading />", () => {
  it("should render with default values", () => {
    renderTheme(<Heading>Text</Heading>);
    const heading = screen.getByRole("heading", { name: "Text" });

    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      "font-size": theme.font.sizes.xhuge,
      "text-transform": "none",
    });

    expect(heading).toMatchSnapshot();
  });

  it("should render in UPPERCASE", () => {
    renderTheme(<Heading uppercase={true}>Text</Heading>);
    const heading = screen.getByRole("heading", { name: "Text" });

    expect(heading).toHaveStyle({
      "text-transform": "uppercase",
    });
    expect(heading).toMatchSnapshot();
  });

  it("should render with white color", () => {
    renderTheme(<Heading colorDark={false}>Text</Heading>);
    const heading = screen.getByRole("heading", { name: "Text" });

    expect(heading).toHaveStyle({
      color: theme.colors.lightGrayColor,
      "font-size": theme.font.sizes.xhuge,
      "text-transform": "none",
    });
    expect(heading).toMatchSnapshot();
  });
  // Sizes
  it("should render heading in small size", () => {
    renderTheme(<Heading size="small">Text</Heading>);
    const heading = screen.getByRole("heading", { name: "Text" });

    expect(heading).toHaveStyle({
      "font-size": theme.font.sizes.medium,
    });
    expect(heading).toMatchSnapshot();
  });

  it("should render heading in medium size", () => {
    renderTheme(<Heading size="medium">Text</Heading>);
    const heading = screen.getByRole("heading", { name: "Text" });

    expect(heading).toHaveStyle({
      "font-size": theme.font.sizes.large,
    });
    expect(heading).toMatchSnapshot();
  });

  it("should render heading in large size", () => {
    renderTheme(<Heading size="large">Text</Heading>);
    const heading = screen.getByRole("heading", { name: "Text" });

    expect(heading).toHaveStyle({
      "font-size": theme.font.sizes.xlarge,
    });
    expect(heading).toMatchSnapshot();
  });

  it("should render heading in small size", () => {
    renderTheme(<Heading size="huge">Text</Heading>);
    const heading = screen.getByRole("heading", { name: "Text" });

    expect(heading).toHaveStyle({
      "font-size": theme.font.sizes.xhuge,
    });
    expect(heading).toMatchSnapshot();
  });
  //@media
  it("should render correct font-size (large) when the screen is less than 768px", () => {
    renderTheme(<Heading size="huge">Text</Heading>);
    const heading = screen.getByRole("heading", { name: "Text" });

    expect(heading).toHaveStyleRule("font-size", theme.font.sizes.xlarge, {
      media: theme.media.lteMedium,
    });
    expect(heading).toMatchSnapshot();
  });

  it("should render the correct html heading element", () => {
    const { container } = renderTheme(<Heading as={"h6"}>Text</Heading>);
    const h6 = container.querySelector("h6");

    expect(h6.tagName.toLowerCase()).toBe("h6");
    expect(h6).toMatchSnapshot();
  });
});
