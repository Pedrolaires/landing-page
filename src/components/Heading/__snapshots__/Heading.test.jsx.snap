// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render correct font-size (large) when the screen is less than 768px 1`] = `
.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
>
  Text
</h1>
`;

exports[`<Heading /> should render heading in large size 1`] = `
.c0 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: none;
}

<h1
  class="c0"
>
  Text
</h1>
`;

exports[`<Heading /> should render heading in medium size 1`] = `
.c0 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

<h1
  class="c0"
>
  Text
</h1>
`;

exports[`<Heading /> should render heading in small size 1`] = `
.c0 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: none;
}

<h1
  class="c0"
>
  Text
</h1>
`;

exports[`<Heading /> should render heading in small size 2`] = `
.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
>
  Text
</h1>
`;

exports[`<Heading /> should render in UPPERCASE 1`] = `
.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
>
  Text
</h1>
`;

exports[`<Heading /> should render the correct html heading element 1`] = `
.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h6
  class="c0"
>
  Text
</h6>
`;

exports[`<Heading /> should render with default values 1`] = `
.c0 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
>
  Text
</h1>
`;

exports[`<Heading /> should render with white color 1`] = `
.c0 {
  color: #EEE;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
>
  Text
</h1>
`;
